pkg/asset/installconfig/gcp/dns.go

```diff
@@ -29,6 +31,49 @@ func GetPublicZone(ctx context.Context, project, baseDomain string) (*dns.Manage
 	return dnsZone, nil
 }
 
+// ParseZone will attempt to extract the project and zone from a long zone name that
+// includes the full path (i.e. https://dns.googleapis.com/dns/...). If the data cannot be
+// parsed or does not include this information, the default (provided) values will be returned
+func ParseZone(defaultProject, providedZoneName string) (string, string, error) {
+	splitZoneName := func(fullySpecifiedZone string) (string, string) {
+		splitZoneName := strings.Split(fullySpecifiedZone, "/")
+		return splitZoneName[len(splitZoneName)-3], splitZoneName[len(splitZoneName)-1]
+	}
+
+	zoneName := providedZoneName
+	project := defaultProject
+
+	matched, err := regexp.MatchString(`^https://dns\.googleapis\.com/dns/`, zoneName)
+	if err != nil {
+		return "", "", err
+	} else {
+		if matched {
+			project, zoneName = splitZoneName(zoneName)
+		}
+	}
+
+	return project, zoneName, nil
+}
```

Comment:
The "https://" would be a URL as opposed to a "full resource name". Most users do not interact with the resource URLs... so I hesitate to use them here.

https://cloud.google.com/apis/design/resource_names#resource_name_vs_url

However, I don't see where DNS resources are defined to have full resource names.

https://cloud.google.com/iam/docs/full-resource-names

Perhaps we should instead expect a partial resource name. This removes the need for us to assume the format of a full resource name for a managedZone.

projects/project-id/managedZones/managed-zone-name

We should ensure the name matches this pattern precisely (has both the "projects/" and "/managedZones/" sections) before assuming it is a partial resource name. Otherwise, we just send the entire string to the gcp api for validation as-is.